contrib/format-hdf5/src/main/java/org/apache/drill/exec/store/hdf5/writers/HDF5DataWriter.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.exec.store.hdf5.writers;
+
+import ch.systemsx.cisd.hdf5.IHDF5Reader;
+import org.apache.drill.exec.physical.resultSet.RowSetLoader;
+
+import java.util.ArrayList;
+
+public class HDF5DataWriter {
+  final RowSetLoader columnWriter;
+
+  final IHDF5Reader reader;
+
+  final String datapath;
+
+  String fieldName;
+
+  int colCount;
+
+  int counter;
+
+  Object[][] compoundData;
+
+  public HDF5DataWriter(IHDF5Reader reader, RowSetLoader columnWriter, String datapath) {
+    this.reader = reader;
+    this.columnWriter = columnWriter;
+    this.datapath = datapath;
+    counter = 0;
+  }
+
+  public HDF5DataWriter(IHDF5Reader reader, RowSetLoader columnWriter, String datapath, String fieldName, int colCount) {
+    this.reader = reader;
+    this.columnWriter = columnWriter;
+    this.datapath = datapath;
+    this.fieldName = fieldName;
+    this.colCount = colCount;
+    counter = 0;
+  }
+
+  public boolean write() {
+    return false;
+  }
+
+  public boolean hasNext() {
+    return false;
+  }
+
+  public int currentRowCount() {
+    return counter;
+  }
+
+  public ArrayList getColumn(int columnIndex) {
```

Comment:
Need to parameterize, at least `ArrayList<?>` or `ArrayList<Object>`. Actually, better to use the base type: `List<Object>`.

Reply:
Fixed.